### PR TITLE
Refactor/fix duplicate issue check to use open/closed and ineligible

### PR DESCRIPTION
**Pull request by shanear**
connects #9875

### CODE REVIEW

File: app/models/contestable_issue.rb [hunk 1]
```diff
@@ -102,37 +102,31 @@ def find_latest_contestable_issues
...
-
-    return unless different_decision_review(found_request_issue)
-
-    found_request_issue
-  end
-
-  def different_decision_review(found_request_issue)
-    return unless found_request_issue
-
-    found_request_issue.decision_review_id != contesting_decision_review.id ||
```
Comment (shanear) [inline app/models/contestable_issue.rb]: This is redundant of the scope here: 
https://github.com/department-of-veterans-affairs/caseflow/pull/9893/files#diff-703856fcb983e2497bfd75cf7ffd77afR126

File: app/models/contestable_issue.rb [hunk 2]
```diff
@@ -102,37 +102,31 @@ def find_latest_contestable_issues
...
 
   def conflicting_request_issue_by_decision_issue
     return unless decision_issue&.id
 
-    potentially_conflicting_request_issues.find_active_by_contested_decision_id(decision_issue.id)
+    potentially_conflicting_request_issues.find_by(contested_decision_issue_id: decision_issue.id)
   end
 
   def potentially_conflicting_request_issues
-    RequestIssue.where.not(decision_review: contesting_decision_review)
```
Comment (shanear) [inline app/models/contestable_issue.rb]: The scope here was previously not getting used. The logic here was being implemented by:
https://github.com/department-of-veterans-affairs/caseflow/pull/9893/files#diff-703856fcb983e2497bfd75cf7ffd77afL131

File: app/models/request_issue.rb [hunk 3]
```diff
@@ -153,7 +153,15 @@ def not_deleted
       where.not(decision_review_id: nil)
     end
 
+    def eligible
```
Comment (leikkisa) [inline app/models/request_issue.rb]: I know these are little, but I believe there's a place where these scopes have tests, might be good to add one here.

File: app/models/contestable_issue.rb [hunk 4]
```diff
@@ -102,37 +102,31 @@ def find_latest_contestable_issues
...
 
+  def conflicting_request_issue
+    conflicting_request_issue_by_decision_issue || conflicting_request_issue_by_rating
+  end
+
   def conflicting_request_issue_by_rating
     return unless rating_issue_reference_id
 
-    potentially_conflicting_request_issues.find_active_by_contested_rating_issue_reference_id(rating_issue_reference_id)
+    potentially_conflicting_request_issues.find_by(contested_rating_issue_reference_id: rating_issue_reference_id)
```
[the remark this answers is not in the document]
Comment (shanear) [reply app/models/contestable_issue.rb]: `potentially_conflicting_request_issues` does scope it to open issues

File: app/models/request_issue.rb (hunk 3 above)
Comment (shanear) [reply app/models/request_issue.rb]: yeah, i have a longer way to go here, I'm gonna add tests for these scopes in a follow up PR.

**Pull request merged**